### PR TITLE
revert back to the old JSX transform

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

This reverts the new JSX transform added in #2102 that shipped with 3.12.0.

This is because we immediately started receiving [reports](https://github.com/iTwin/viewer/issues/318) that it doesn't work for React 17 users that use Webpack 5. I did some digging and found that the React team was made aware of [this issue back in 2020](https://redirect.github.com/facebook/react/issues/20235), and they willingly decided not to fix it until v18 (bad decision from them imo). And yes, this means their [lofty claims](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#how-to-upgrade-to-the-new-jsx-transform) about the new transform being supported in older React versions are untrue.

The main reason for using the new JSX transform was that it was ["required"](https://react.dev/blog/2024/04/25/react-19-upgrade-guide#installing) for React 19. However, upon reading closely, it seems like it will only raise a warning, not an error, so it is in fact *not* "required". Further, I tested iTwinUI 3.11.3 (which uses the old JSX transform) with React 19 RC and did not even see any warnings.

This does mean that we are stuck with the older, less performant JSX runtime for the foreseeable future. I'm making this change reluctantly, but it seems to work in React 17-19 and is the least frustrating outcome for our users (the majority of whom are still using React 17). I also tried to find a "best of both worlds" solution for the last couple days, with no luck.

I've strategically made the smallest change possible instead of fully reverting #2102. Only two source files (`StatusIconMap.tsx` and `selectionColumn.tsx`) needed to be changed to add a missing `React` import. Everything else (mainly test files) continue to use the new transform, since these are not relevant for our users.

## Testing

Veridied that the build output now contains `React.createElement` everywhere.

## Docs

N/A

### CODE REVIEW

**Comment by Ben-Pusey-Bentley** [comment]
I think the `vitest.config.ts` file needs `esbuild.jsx` to be set to `'auotmatic'` in order to build our `test.tsx` unit test files with the new JSX transform. Docs: [https://vitejs.dev/config/shared-options.html#esbuild](https://vitejs.dev/config/shared-options.html#esbuild) [https://esbuild.github.io/api/#jsx](https://esbuild.github.io/api/#jsx).

**Review by mayank99 — COMMENTED** [review]
> I think the `vitest.config.ts` file needs `esbuild.jsx` to be set to `'auotmatic'` in order to build our `test.tsx` unit test files with the new JSX transform.

Thanks, fixed. I was trying to figure out how to make it use the `tsconfig.test.json` file but it seems like that's not possible.

File: packages/itwinui-react/.eslintrc.cjs
Comment (mayank99) [inline packages/itwinui-react/.eslintrc.cjs:1]: The changes in this file basically re-enable the old React linter rules for this specific workspace, while ignoring test files.

I found that the eslintignore file was not working, so I deleted it. I also found that the `Combobox.test-types` file was named incorrectly, so I renamed it (maybe I should do that in a separate PR tho).

Comment (Ben-Pusey-Bentley) [reply packages/itwinui-react/.eslintrc.cjs:1]: Yeah, it might make sense to move the file rename to a separate PR, just because it isn't really related to the issue being addressed in this PR. But it is a small enough change that I think it makes sense to just add it to the PR description. Your call.

**Review by Ben-Pusey-Bentley — APPROVED** [review]
Nit: The Docs section of the PR description should acknowledge the changeset. Otherwise, LGTM.

File: packages/itwinui-react/src/utils/icons/StatusIconMap.tsx [hunk 1]
```diff
@@ -2,17 +2,23 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
 import { SvgInfoCircular } from './SvgInfoCircular.js';
 import { SvgStatusError } from './SvgStatusError.js';
 import { SvgStatusSuccess } from './SvgStatusSuccess.js';
 import { SvgStatusWarning } from './SvgStatusWarning.js';
-import type { CommonProps } from '../props.js';
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/src/utils/icons/StatusIconMap.tsx:9]: Yes and no. I initially made this change because I was getting "unused `React` import" errors. I ended up fixing the config to not raise these errors, but I left this change because `CommonProps` should not be used anyway.

File: packages/itwinui-react/.eslintrc.cjs [hunk 2]
```diff
@@ -13,5 +13,8 @@ module.exports = {
   rules: {
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/.eslintrc.cjs:13]: No, we only want the old JSX transform in the component source. We can use the new transform everywhere else.

File: packages/itwinui-react/tsconfig.test.json [hunk 3]
```diff
@@ -1,5 +1,8 @@
 {
   "extends": "./tsconfig.json",
-  // TODO: Test all the files, and not just the Table files or just the build files.
-  "include": ["src/**/*.test.*", "src/**/*.types-test.*"]
+  "compilerOptions": {
+    "jsx": "react-jsx",
```
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-react/tsconfig.test.json:4]: Yes. As mentioned in the PR description, I didn't want to revert all the test files to the old transform because these are not relevant for our users.

File: packages/itwinui-react/.eslintrc.cjs
Comment (mayank99) [reply packages/itwinui-react/.eslintrc.cjs:1]: I'll keep it here to speed things up, since this is blocking release
